sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NewGuidShouldNotBeUsed.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class Program
+    {
+        public void Foo()
+        {
+            var result = new Guid(); // Noncompliant {{Please review this use of 'new Guid()'.}}
+//                       ^^^^^^^^^^
+
+            result = Guid.Empty;
+            result = Guid.NewGuid();
+            result = new Guid(new byte[0]);
```

Comment:
This raises a runtime argument exception saying byte array should have a length of exactly 16 bytes. Looks like a good candidate for another rule.